python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1363,21 +1363,17 @@ def _log_materialization_planned_event_for_asset(
         from dagster._core.definitions.partition_key_range import PartitionKeyRange
         from dagster._core.events import AssetMaterializationPlannedData, DagsterEvent
 
+        is_partitioned = check.not_none(output.properties).is_asset_partitioned
         partition_tag = dagster_run.tags.get(PARTITION_NAME_TAG)
         partition_range_start, partition_range_end = (
             dagster_run.tags.get(ASSET_PARTITION_RANGE_START_TAG),
             dagster_run.tags.get(ASSET_PARTITION_RANGE_END_TAG),
         )
 
-        if partition_tag and (partition_range_start or partition_range_end):
-            raise DagsterInvariantViolationError(
-                f"Cannot have {ASSET_PARTITION_RANGE_START_TAG} or"
-                f" {ASSET_PARTITION_RANGE_END_TAG} set along with"
-                f" {PARTITION_NAME_TAG}"
-            )
-
-        partitions_subset = None
-        if partition_range_start or partition_range_end:
+        if is_partitioned and partition_tag:
+            partition = partition_tag
+            partitions_subset = None
```

Comment:
`AssetMaterializationPlannedData` accepts one non-null value for `partition` and `partitions_subset`. Prefer `partition` if we are targeting a single partition.